src/Sylius/Bundle/CoreBundle/Controller/OrderController.php

```diff
@@ -53,4 +59,101 @@ public function thankYouAction(Request $request)
 
         return $this->viewHandler->handle($configuration, $view);
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function showAvailableShippingMethodsAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        /** @var OrderInterface $cart */
+        $cart = $this->getCartOr404($request->attributes->get('orderId'));
+
+        if (!$this->getCartStateMachine($cart, 'sylius_order_checkout')->can('select_shipping')) {
+            throw new BadRequestHttpException('The shipment methods cannot be resolved in the current state of cart!');
+        }
+
+        $shipments = [];
+
+        foreach ($cart->getShipments() as $shipment) {
+            $shipments['shipments'][] = [
+                'methods' => $this->getShippingMethodsResolver()->getSupportedMethods($shipment),
+            ];
+        }
+
+        return $this->viewHandler->handle($configuration, View::create($shipments));
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function showAvailablePaymentMethodsAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        /** @var OrderInterface $cart */
+        $cart = $this->getCartOr404($request->attributes->get('orderId'));
```

Comment:
These are two variable assignments, which I can just put without a blank line in the middle. But I would prefer to not extract them more.
Also I don't want to extract more logic, as it could probable make more harm than good.